### PR TITLE
Dev/mookerem/arm64x

### PR DESCRIPTION
**Pull request by moyo1997**
Changes needed to support ARM64X for windows team. 
The changes are only made to the VS generator

### CODE REVIEW

File: Modules/Platform/Windows-MSVC.cmake [hunk 1]
```diff
@@ -242,6 +242,10 @@ else()
     set(_FLAGS_CXX "${_GR} /GX")
     set(CMAKE_C_STANDARD_LIBRARIES_INIT "kernel32.lib user32.lib gdi32.lib winspool.lib comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib uuid.lib odbc32.lib odbccp32.lib")
   endif()
+  
+  if((_MSVC_C_ARCHITECTURE_FAMILY STREQUAL "ARM64EC") OR (_MSVC_CXX_ARCHITECTURE_FAMILY STREQUAL "ARM64EC"))
+    string(APPEND CMAKE_C_STANDARD_LIBRARIES_INIT " softintrin.lib")
```
Comment (sfreed141) [inline Modules/Platform/Windows-MSVC.cmake:247]: Is there an equivalent `CMAKE_CXX_STANDARD_LIBRARIES_INIT` that should be appended to?

File: Source/cmGlobalVisualStudio8Generator.cxx [hunk 2]
```diff
@@ -227,9 +227,14 @@ void cmGlobalVisualStudio8Generator::WriteSolutionConfigurations(
   std::ostream& fout, std::vector<std::string> const& configs)
 {
   fout << "\tGlobalSection(SolutionConfigurationPlatforms) = preSolution\n";
-  for (std::string const& i : configs) {
-    fout << "\t\t" << i << "|" << this->GetPlatformName() << " = " << i << "|"
-         << this->GetPlatformName() << "\n";
+   std::vector<std::string> platforms = { this->GetPlatformName() };
```
Comment (sfreed141) [inline Source/cmGlobalVisualStudio8Generator.cxx]: nit: remove the extra space at beginning of the line

File: Source/cmGlobalVisualStudio8Generator.cxx [hunk 3]
```diff
@@ -227,9 +227,14 @@ void cmGlobalVisualStudio8Generator::WriteSolutionConfigurations(
...
+  if (this->GetBuildAsX())
+    platforms.push_back("ARM64");
```
Comment (sfreed141) [inline Source/cmGlobalVisualStudio8Generator.cxx:232]: should this be ARM64X?

Also nit: add braces around statement body (although I'm not 100% sure if that's part of the official CMake style guidelines)

File: Source/cmGlobalVisualStudio8Generator.cxx [hunk 4]
```diff
@@ -241,38 +246,43 @@ void cmGlobalVisualStudio8Generator::WriteProjectConfigurations(
...
-    const char* dstConfig = i.c_str();
-    if (target.GetProperty("EXTERNAL_MSPROJECT")) {
-      if (cmProp m = target.GetProperty("MAP_IMPORTED_CONFIG_" +
-                                        cmSystemTools::UpperCase(i))) {
-        cmExpandList(*m, mapConfig);
-        if (!mapConfig.empty()) {
-          dstConfig = mapConfig[0].c_str();
+  std::vector<std::string> platforms = { this->GetPlatformName() };
+  if (this->GetBuildAsX())
+    platforms.push_back("ARM64");
```
Comment (sfreed141) [inline Source/cmGlobalVisualStudio8Generator.cxx:252]: same comments as above

File: Source/cmake.cxx [hunk 5]
```diff
@@ -604,6 +604,10 @@ void cmake::ProcessCacheArg(const std::string& var, const std::string& value,
     }
   }
 
+  if (var == "MSVC_BUILD_AS_X") {
+    this->BuildAsX = true;
+	}
```
Comment (sfreed141) [inline Source/cmake.cxx]: nit: fix indentation

**Review by sfreed141 — APPROVED** [review]
lgtm! I just left a few questions and nits. If the PR doesn't require any changes I'm fine merging without fixing the nits though.

File: Modules/Platform/Windows-MSVC.cmake (hunk 1 above)
Comment (moyo1997) [reply Modules/Platform/Windows-MSVC.cmake:247]: CMAKE_CXX_STANDARD_LIBRARIES_INIT  is set to CMAKE_C_STANDARD_LIBRARIES_INIT

File: Source/cmGlobalVisualStudio8Generator.cxx (hunk 3 above)
Comment (moyo1997) [reply Source/cmGlobalVisualStudio8Generator.cxx:232]: This should be arm64

**Pull request merged**